google/cloud/storage/blob.py

```diff
@@ -1838,10 +1937,16 @@ def _do_resumable_upload(
             if_metageneration_match=if_metageneration_match,
             if_metageneration_not_match=if_metageneration_not_match,
             timeout=timeout,
+            checksum=checksum,
         )
 
         while not upload.finished:
-            response = upload.transmit_next_chunk(transport, timeout=timeout)
+            try:
+                response = upload.transmit_next_chunk(transport, timeout=timeout)
+            except resumable_media.DataCorruption:
+                # Attempt to delete the corrupted object.
+                self.delete()
```

Comment:
If this fails for any reason, both exceptions will be reported, using the Python exception pattern "During handling of the above exception, another exception occurred"

Reply:
What occurs here if a valid previous generation of the blob exists?